### PR TITLE
GEOMESA-2814 Configure GitHub Actions

### PR DESCRIPTION
**Pull request by jeffklassen**
Signed-off-by: Jeffrey Klassen <jeffrey.s.klassen@gmail.com>

### CODE REVIEW

File: .github/workflows/build-and-test.yml [hunk 1]
```diff
@@ -0,0 +1,30 @@
+name: CI
+
+on: [push]
```
Comment (elahrvivaz) [inline .github/workflows/build-and-test.yml]: we should probably make this push or pr

File: .github/workflows/build-and-test.yml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+        with:
+          java-version: 1.8
+      - uses: actions/cache@v1
+        with:
+          path: ~/.m2/repository
+          key: ${{ runner.os }}-maven-${{ hashFiles('**/pom.xml') }}
+          restore-keys: |
+                ${{ runner.os }}-maven-
+      - name: Build with Maven
+        run: mvn clean install -DskipTests -T4 2>&1| grep -e 'Building GeoMesa' -e '\(maven-surefire-plugin\|maven-jar-plugin\|scala-maven-plugin.*:compile\)'
```
Comment (elahrvivaz) [inline .github/workflows/build-and-test.yml]: we could use build/mvn and possibly cache the downloads of zinc if that's easy

**Comment by elahrvivaz** [comment]
this looks good - the only concern i have is that we won't be able to see the errors if it fails, due to the grepping of the output. the travis build `tee`d to a log file and then dumped the last 500 lines of the log file on a failure - we could do that, or just remove the grep.

File: .github/workflows/build-and-test.yml (hunk 1 above)
Comment (jeffklassen) [reply .github/workflows/build-and-test.yml]: done

File: .github/workflows/build-and-test.yml (hunk 2 above)
Comment (jeffklassen) [reply .github/workflows/build-and-test.yml]: @elahrvivaz implemented. I've seen that caching is not mega fast, while the zinc download is. I'm going to investigate this w/o the cached zinc download

**Comment by elahrvivaz** [comment]
merged as fb633e31d98e68eac358404a843dea3fde086240

**Pull request closed without merging**